accounts-db/src/accounts_db.rs

```diff
@@ -4006,6 +4020,18 @@ impl AccountsDb {
                         unrefed_pubkeys.push(pubkey);
                         result = AccountsIndexScanResult::Unref;
                         dead += 1;
+
+                        // If we are marking something dead, and the only remaining alive account is zero lamport, then make that zero lamport slot ready to be cleaned.
+                        // If that slot happens to only contain zero lamport accounts, the whole slot will go away
+                        if slot_list.len() == 1 // should we also check for ref counts here?
```

Comment:
Is it possible that the slot_list contains more than one zero lamport accounts?
Can we relax the requirement for slot_list.len() ==1?